### PR TITLE
refactor: FeelEngineBuilder for Java usage

### PR DESCRIPTION
**Pull request by saig0**
## Description

Follow-up of #832.

- Move the Java builder method to the root 
- Add JavaDoc
- Extend the test cases for the Java builder

## Related issues

related to #724

### CODE REVIEW

File: src/main/scala/org/camunda/feel/api/FeelEngineBuilder.scala [hunk 1]
```diff
@@ -82,5 +78,18 @@ case class FeelEngineBuilder private (
...
+    */
   def create(): FeelEngineBuilder = FeelEngineBuilder()
+
+  /** Creates a new preconfigured builder for the FEEL engine. Use it if the engine is called from
+    * Java code.
+    *
+    * @return
+    *   a new builder
+    */
+  def forJava(): FeelEngineBuilder = FeelEngineBuilder().withCustomValueMapper(new JavaValueMapper)
```
Comment (megglos) [inline src/main/scala/org/camunda/feel/api/FeelEngineBuilder.scala:94]: ❓ As a Scala newbie, this is kind of a cleanup to not make the utility method part of the actual builder interface but expose it on the singleton object directly?

File: src/test/java/org/camunda/feel/api/FeelEngineBuilderJava.java [hunk 2]
```diff
@@ -17,7 +17,8 @@
 package org.camunda.feel.api;
 
 public class FeelEngineBuilderJava {
-    public FeelEngineApi buildEngine() {
-        return FeelEngineBuilder.create().forJava().build();
-    }
+
+  public FeelEngineApi buildEngine() {
+    return FeelEngineBuilder.forJava().build();
```
Comment (megglos) [inline src/test/java/org/camunda/feel/api/FeelEngineBuilderJava.java:22]: that answers my question, I guess

File: src/test/scala/org/camunda/feel/api/FeelEngineBuilderTest.scala [hunk 3]
```diff
@@ -21,9 +21,18 @@ import org.scalatest.matchers.should.Matchers
...
-  "A FeelEngineBuilder" should "successfully build a FeelEngineApi instance" in {
+  "The FeelEngineBuilder" should "be usable from Java" in {
     val javaBuilder = new FeelEngineBuilderJava()
-    val engine: FeelEngineApi = javaBuilder.buildEngine()
-    engine should not be null
+
+    javaBuilder.buildEngine() shouldBe a[FeelEngineApi]
+  }
+
+  it should "build a preconfigured Java engine" in {
```
Comment (megglos) [inline src/test/scala/org/camunda/feel/api/FeelEngineBuilderTest.scala:30]: 🔧 
```suggestion
  it should "build a preconfigured Java engine and evaluate expression" in {
```

**Review by megglos — APPROVED** [review]
LGTM!

File: src/main/scala/org/camunda/feel/api/FeelEngineBuilder.scala (hunk 1 above)
Comment (saig0) [reply src/main/scala/org/camunda/feel/api/FeelEngineBuilder.scala:94]: Yes. :+1:

**Pull request merged**